### PR TITLE
CARL provides the CA certificate within a PeerSetup to increase usability

### PR DESCRIPTION
**Pull request by mtwardawski**
added ca certificate to carls config files carl.toml and carl-development.toml, carl uses that ca certificate to generate peer setup. Decocec Ca Certificate was therefore added to setup string. Resolved a clippy finding in authenticated.rs.

### CODE REVIEW

File: opendut-carl/src/grpc/peer_manager.rs [hunk 1]
```diff
@@ -20,15 +21,17 @@ pub struct PeerManagerFacade {
     resources_manager: ResourcesManagerRef,
     vpn: Vpn,
     carl_url: Url,
+    ca: String,
 }
 
 impl PeerManagerFacade {
 
-    pub fn new(resources_manager: ResourcesManagerRef, vpn: Vpn, carl_url: Url) -> Self {
+    pub fn new(resources_manager: ResourcesManagerRef, vpn: Vpn, carl_url: Url, ca: String) -> Self {
```
Comment (kKdH) [inline opendut-carl/src/grpc/peer_manager.rs:29]: use new-type instead of `String`

File: opendut-carl/src/grpc/peer_manager.rs [hunk 2]
```diff
@@ -233,7 +238,12 @@ mod tests {
     async fn test_successful_create_delete() -> Result<()> {
 
         let resources_manager = Arc::new(ResourcesManager::new());
-        let testee = PeerManagerFacade::new(Arc::clone(&resources_manager), Vpn::Disabled, Url::parse("https://example.com:1234").unwrap());
+        let testee = PeerManagerFacade::new(
+            Arc::clone(&resources_manager),
+            Vpn::Disabled,
+            Url::parse("https://example.com:1234").unwrap(),
+            "I'm a ca cert string".to_string()
```
Comment (kKdH) [inline opendut-carl/src/grpc/peer_manager.rs:245]: use `include_str!()` and include `resources/development/tls/insecure-development-ca.pem`

File: opendut-carl/src/grpc/peer_manager.rs [hunk 3]
```diff
@@ -338,7 +353,12 @@ mod tests {
     async fn unregister_fails_when_no_id_specified() -> Result<()> {
 
         let resources_manager = Arc::new(ResourcesManager::new());
-        let testee = PeerManagerFacade::new(Arc::clone(&resources_manager), Vpn::Disabled, Url::parse("https://example.com:1234").unwrap());
+        let testee = PeerManagerFacade::new(
+            Arc::clone(&resources_manager),
+            Vpn::Disabled,
+            Url::parse("https://example.com:1234").unwrap(),
+            "I'm a ca cert string".to_string()
```
Comment (kKdH) [inline opendut-carl/src/grpc/peer_manager.rs:360]: same as above

File: opendut-types/proto/opendut/types/util/net.proto [hunk 4]
```diff
@@ -6,6 +6,10 @@ message Url {
   string value = 1;
 }
 
+message Certificate {
+  string value = 1;
```
Comment (kKdH) [inline opendut-types/proto/opendut/types/util/net.proto:10]: use `bytes` as datatype

File: opendut-types/src/util/net.rs [hunk 5]
```diff
@@ -51,3 +51,32 @@ pub enum NetworkInterfaceNameError {
     #[error("Due to operating system limitations, the name for network interfaces may not be longer than {max} characters!")]
     TooLong { value: String, max: usize }
 }
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[serde(transparent)]
+pub struct Certificate(pub String);
```
Comment (kKdH) [inline opendut-types/src/util/net.rs:57]: Maybe we can replace our new-type with a 'real' certificate type for example from rustls.

**Review by kKdH — CHANGES_REQUESTED** [review]
Thank you for your contribution! ❤

**Review by reimarstier — APPROVED** [review]
LGTM

**Pull request merged**